Remove sunweg integration

rokam opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->
Removing Sunweg integration due to authentication with CAPTCHA. The integration can't work without authentication.


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
https://github.com/rokam/sunweg/issues/13


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [x] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

joostlek reviewed (commented):
So can you please elaborate on how it used to work and what happened to the API

rokam commented:
There is no documented API and it was using the main page authentication form to retrieve the auth token. But sunweg just added a recaptcha to login page. https://sun.weg.net/sign-in

I tried to contact them for sometime without a response for a month. So, there is nothing that can be done ATM.

joostlek commented:
And is there a way we can still use the API via a token we can fetch via the browser?

rokam commented:
Yes, but the token will expire in some days and there is no refresh token method.

joostlek commented:
Right, I will discuss with others, I think it will be okay to remove it. But since this is a config entry we should raise an issue. Please check integrations like `myq`, `life360` or `mazda`. They help the user to gracefully show the user to remove it.

frenck commented:
@joostlek @rokam What is the state on this one?

../Frenck

joostlek commented:
Right, we still need to get on with this. I am not sure why the CI cancelled the jobs

rokam commented:
> @joostlek @rokam What is the state on this one?
> 
> ../Frenck

Looks ready for me.

rokam committed:
chore(sunweg): remove sunweg integration

rokam commented:
rebased

abmantis commented on homeassistant/components/sunweg/config_flow.py:
can't this file be removed?

rokam replied:
I did follow those as example `myq`, `life360` or `mazda`

abmantis commented on homeassistant/components/sunweg/manifest.json:
```diff
@@ -1,10 +1,10 @@
   "config_flow": true,
   "documentation": "https://www.home-assistant.io/integrations/sunweg",
   "iot_class": "cloud_polling",
   "loggers": ["sunweg"],
```
```suggestion
  "loggers": [],
```

abmantis commented on homeassistant/components/sunweg/strings.json:
```diff
@@ -1,35 +1,8 @@
+  "issues": {
+    "integration_removed": {
+      "title": "The SunWEG integration has been removed",
+      "description": "The SunWEG integration has been removed from Home Assistant.\n\nThe library that Home Assistant uses to connect with their services, [don't work anymore as expected, demanding daily token renew]({issue}).\n\nTo resolve this issue, please remove the (now defunct) integration entries from your Home Assistant setup. [Click here to see your existing SunWEG integration entries]({ ...
```
```suggestion
      "description": "The SunWEG integration has been removed from Home Assistant.\n\nThe library that Home Assistant uses to connect with SunWEG services, [doesn't work as expected anymore, demanding daily token renew]({issue}).\n\nTo resolve this issue, please remove the (now defunct) integration entries from your Home Assistant setup. [Click here to see your existing SunWEG integration entries]({entries})."
```

abmantis commented:
@joostlek is it fine to proceed with the removal?

rokam committed:
Update homeassistant/components/sunweg/strings.json

Co-authored-by: Abílio Costa <abmantis@users.noreply.github.com>

rokam committed:
Merge branch 'dev' into rem-sunweg

abmantis commented:
There are still 2 unresolved comments.

abmantis commented:
@rokam I think you missed this one: https://github.com/home-assistant/core/pull/124230#discussion_r1992046863

rokam committed:
Update homeassistant/components/sunweg/manifest.json

Co-authored-by: Abílio Costa <abmantis@users.noreply.github.com>

rokam committed:
Merge branch 'dev' into rem-sunweg

rokam committed:
feat: added async remove entry

rokam commented:
> @rokam I think you missed this one: [#124230 (comment)](https://github.com/home-assistant/core/pull/124230#discussion_r1992046863)
> 
> Also, should we implement `async_remove_entry` like the `myq` integration does, so that we automatically remove disabled and ignored entries?

I did as requested, although I think that the unload was enought.

rokam committed:
Merge branch 'dev' into rem-sunweg

abmantis reviewed (approved):
LGTM, but I am not sure we still need `async_unload_entry`. `async_remove_entry` should be enough

joostlek commented:
I think you are right, we can remove the unload function

abmantis committed:
Clean setup_entry; add tests

abmantis commented:
@rokam  I've removed the code from `async_unload_entry` and added tests for the repair issue and entry removal. Let me know if it is fine for you so we can merge this.

rokam commented:
> @rokam  I've removed the code from `async_unload_entry` and added tests for the repair issue and entry removal. Let me know if it is fine for you so we can merge this.

Lgtm

MartinHjelmare commented:
There's a merge conflict.

rokam committed:
Merge branch 'dev' into rem-sunweg

abmantis reviewed (approved):
Thanks!
